lib/CdeContext.tsx

```diff
@@ -1,83 +1,159 @@
 }>({
-    labName: '',
+
+    name: '',
+    datasetSample: [],
+    datasetMapping: {},
+    datasetMappingHeader: [],
+    collections: [],
     config: {
         width: "100%",
         height: "100%",
     },
-    inputMappings: [],
-    mapping: {} as DatasetCDEMapping,
-    setMapping: () => {},
+
     step: 0,
-    setStep: () => {},
+    setStep: () => {
+    },
     loadingMessage: null,
-    setLoadingMessage: () => {},
+    setLoadingMessage: () => {
+    },
     errorMessage: null,
-    setErrorMessage: () => {},
-    isOpen: true,
-    handleClose: () => { },
-    infoOpen: false,
-    setInfoOpen: () => {}
+    setErrorMessage: () => {
+    },
+    handleClose: () => {
+    },
 });
 
 export const useCdeContext = () => useContext(CdeContext);
 
-export const CdeContextProvider = ({children, labName, callback, inputMappings, config}: PropsWithChildren<InitParams>) => {
-    const [mapping, setMapping] = useState<DatasetCDEMapping>({});
-    const [step, setStep] = useState<number>(STEPS.HOME);
-    const [loadingMessage, setLoadingMessage] = useState<string | null>(null);
-    const [errorMessage, setErrorMessage] = useState<string | null>(null)
-    const [isOpen, setIsOpen] = useState<boolean>(true)
-    const [infoOpen, setInfoOpen] = useState<boolean>(false)
+export const CdeContextProvider = ({
+                                       datasetSample: rawDatasetSample,
+                                       datasetMapping: rawDatasetMapping,
+                                       additionalDatasetMappings: rawAdditionalDatasetMappings,
+                                       collections,
+                                       config,
+                                       name,
+                                       callback,
+                                       children
+                                   }: PropsWithChildren<InitParams>) => {
+
+    let datasetSample;
+    // Process and validate datasetSample
+    try {
+        validateDatasetSample(rawDatasetSample);
+        datasetSample = mapDatasetSample(rawDatasetSample);
+    } catch (error) {
+        const message = error instanceof Error ? error.message : 'An unknown error occurred';
+        // TODO: Check if it's fine to throw or should we show an error in the UI
+        throw new Error(`Invalid dataset sample: ${message}`);
+    }
+
+    // Process and validate datasetMapping
+    let initialDatasetMapping;
+    let initialDatasetMappingHeader;
+    try {
+        validateDatasetMapping(rawDatasetMapping);
+        const datasetMappingData = mapStringTableToDatasetMapping(rawDatasetMapping);
+        initialDatasetMapping = datasetMappingData[0]
+        initialDatasetMappingHeader = datasetMappingData[1]
+    } catch (error) {
+        const message = error instanceof Error ? error.message : 'An unknown error occurred';
+        throw new Error(`Invalid dataset mapping: ${message}`);
+    }
+
+    const additionalDatasetMappings: DatasetMapping[] = [];
```

Comment:
Not part of the context, I'm imagining that we can have a function getSuggestions that uses it and that one will be part of the context. To be implemented later on.